### DISCUSSION TITLE
How to mimic Sublime Text behavior?

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#358] [Q&A] [answered]
On Sublime Text, when you type `{` and enter it will auto expand for you like this:

```
int main(void) {
    // it auto indent for you
}
```

On TextAdept, it's only:

```
{
}
```

**Comment by orbitalquark** [comment, marked as answer]
I have an auto-indent on Enter script in my *init.lua* that may work for 
you:

https://github.com/orbitalquark/.textadept/blob/7294b433375f0e07ee2ba871087addf21129899b/init.lua#L195-L205

**Reply by ghost** [reply]
Yes it worked for me too but only partially. On Sublime Text, on existing function if you place the caret after the beginning bracket and press Enter, it will also do auto indent for you:

```
int main(void) {// move your caret here and press Enter
    // it will also auto indent for you
} // the closing bracket is already there
```

Your code failed to do the same. If the closing bracket is already available, it will not do any auto ident.

**Reply by ghost** [reply]
@orbitalquark Do you have any solutions?

**Comment by orbitalquark** [comment]
Not at this time. I know it can be done, I just didn't have the need for 
it in the example I gave.
